docs/.vale.ini

```diff
@@ -26,11 +26,33 @@ mdx = md
 #    FORMAT-SPECIFIC   #
 ########################
 
-[*.{md,mdx,rst}]
-# Rules in this section are enforced in all md, mdx, and rst files
-
 [*.{md,mdx,rst}]
 BasedOnStyles = Dagster, Terms, Vale
 
-; Ignore all :py directives
-IgnorePatterns = (:py:[^`]+`[^`]+`)
\ No newline at end of file
+; References:
+; - https://vale.sh/docs/topics/scoping/#non-standard-markup
+; - https://github.com/errata-ai/vale/blob/871dafd1e24500cee9d8ad82b25d42a136bb2103/testdata/fixtures/patterns/_vale#L14
+
+; Pattern     : (\\\{[^\n]+\})
+; Regex101    : https://regex101.com/r/GOx8Z6/2
+; Description : Ignore heading anchor renames
+
+; Description : Ignore code snippets
+; Pattern     : (`[^\n^`]+`)
+; Regex101    : https://regex101.com/r/c5EE6S/1
+
+; Pattern     : \[.*\](\(.*\))
+; Regex101    : https://regex101.com/r/GOx8Z6/3
+; Description : Ignore link HREFs
+
+; Additionally, we include TokenIgnores of `<TabItem>` and `<Tabs>` to strip these HTML elements,
+; because when these wrap markdown it causes the markdown linting to fail. For example, code blocks
+; within a tab item.
+
+TokenIgnores = (`[^`]*`), \
+    (\[.*\]\([^)]+\)), \
+    (\\\{[^}]+\}), \
+    (<\/?TabItem.*>), \
+    (<\/?Tabs.*>), \
+    (</?details>), \
+    (<summary>.*<\/summary>)
```

Comment:
@erinkcochran87 / @PedramNavid this should run green now.

Markdown wrapped in `<Tabs>` or `<details>` was not being properly linted (for example, ignoring code blocks).

By including this HTML in the `TokenIgnores`, it is stripped from the document before linting, leaving us with the markdown contents.

This feels very finicky, and I hope we can find something more elegant in the future.